src/java.base/share/classes/sun/security/util/Cache.java

```diff
@@ -265,8 +265,7 @@ public MemoryCache(boolean soft, int maxSize, int lifetime) {
         else
             this.queue = null;
 
-        int buckets = (int)(maxSize / LOAD_FACTOR) + 1;
-        cacheMap = new LinkedHashMap<>(buckets, LOAD_FACTOR, true);
+        cacheMap = new LinkedHashMap<>(1, LOAD_FACTOR, true);
```

Comment:
This looks right. The idea of scaling the initial cache size to the maximum size seems obviously to be wrong.